src/WebJobs.Script.WebHost/Security/KeyManagement/KeyVaultSecretsRepository.cs

```diff
@@ -29,24 +24,29 @@ public class KeyVaultSecretsRepository : BaseSecretsRepository
         private const string FunctionKeyPrefix = "host--functionKey--";
         private const string SystemKeyPrefix = "host--systemKey--";
 
-        private readonly Lazy<KeyVaultClient> _keyVaultClient;
-        private readonly string _vaultName;
+        private readonly Lazy<SecretClient> _secretClient;
         private readonly IEnvironment _environment;
 
-        public KeyVaultSecretsRepository(string secretsSentinelFilePath, string vaultName, string connectionString, ILogger logger, IEnvironment environment) : base(secretsSentinelFilePath, logger, environment)
+        public KeyVaultSecretsRepository(string secretsSentinelFilePath, string vaultName, string clientId, ILogger logger, IEnvironment environment) : base(secretsSentinelFilePath, logger, environment)
         {
-            if (secretsSentinelFilePath == null)
+            if (string.IsNullOrEmpty(secretsSentinelFilePath))
             {
                 throw new ArgumentNullException(nameof(secretsSentinelFilePath));
             }
 
-            _vaultName = vaultName ?? throw new ArgumentNullException(nameof(vaultName));
+            if (string.IsNullOrEmpty(vaultName))
+            {
+                throw new ArgumentNullException(nameof(vaultName));
+            }
+
+            Uri vaultUri = new Uri($"https://{vaultName}{_environment.GetVaultSuffix()}");
```

Comment:
Part of me wonders if we should be asking for `vaultUri` directly instead of asking for the name and constructing the URI.

Reply:
I did this originally, but then I was thinking it might be better to keep the changes to a minimal? Either way works, I don't have a strong preference

Reply:
Using vaultUri